app/controllers/turbo/streams/redirect.rb

```diff
@@ -0,0 +1,25 @@
+module Turbo::Streams::Redirect
+  extend ActiveSupport::Concern
+
+  def redirect_to(options = {}, response_options = {})
+    turbo_frame = response_options.delete(:turbo_frame)
+    turbo_action = response_options.delete(:turbo_action)
+    location = url_for(options)
+
+    if request.format.turbo_stream? && turbo_frame.present?
+      alert, notice, flash_override = response_options.values_at(:alert, :notice, :flash)
+      flash.merge!(flash_override || {alert: alert, notice: notice})
```

Comment:
A possible alternative to handling the `flash` options here could be to call `super` first, _then_ set the response body with the Turbo Stream. This way, the existing `redirect_to` still handles the `flash` options, including any custom flash types added with [`add_flash_types`](https://api.rubyonrails.org/classes/ActionController/Flash/ClassMethods.html).

Similar to the `Turbo::Redirection` module [here](https://github.com/hotwired/turbo-rails/blob/main/UPGRADING.md), we could have something like:
```ruby
def redirect_to(url = {}, options = {})
  if override_turbo_frame?(options)
    super.tap { visit_location_with_turbo_via_turbo_stream(location, options) }
  else
    super
  end
end

private

  def override_turbo_frame?(options)
    options[:turbo_frame].present? && previous_turbo_frame.present?
  end

  def previous_turbo_frame
    request.headers["Turbo-Frame"]
  end

  def visit_location_with_turbo_via_turbo_stream(location, options)
    turbo_frame = options.delete(:turbo_frame)
    turbo_action = options.delete(:turbo_action)

    self.status = 200
    self.response_body = render_to_string(
      "turbo/streams/redirect",
      locals: { location:, turbo_frame:, turbo_action:, previous_turbo_frame: },
      formats: [:turbo_stream]
    )
    response.content_type = "text/vnd.turbo-stream.html"
  end
```